core/src/main/java/org/elasticsearch/cluster/ClusterChangedEvent.java

```diff
@@ -199,10 +199,12 @@ public boolean nodesChanged() {
         return nodesRemoved() || nodesAdded();
     }
 
-    // Determines whether or not the current cluster state represents an entirely
-    // different cluster from the previous cluster state, which will happen when a
-    // master node is elected that has never been part of the cluster before.
-    private boolean isNewCluster() {
+    /**
+     * Determines whether or not the current cluster state represents an entirely
+     * different cluster from the previous cluster state, which will happen when a
+     * master node is elected that has never been part of the cluster before.
```

Comment:
doesn't this also happen when the node first joins the cluster? if fos we should doc it.